src/utilstrencodings.cpp

```diff
@@ -39,6 +39,33 @@ std::string SanitizeString(const std::string& str, int rule)
     return strResult;
 }
 
+bool validateURL(std::string strURL, std::string& strErr, unsigned int maxSize) {
+
+    // Check URL size
+    if (strURL.size() > maxSize) {
+        strErr = strprintf("Invalid URL: %d exceeds limit of %d characters.", strURL.size(), maxSize);
+        return false;
+    }
+
+    std::vector<std::string> reqPre;
+
+    // Required initial strings; URL must contain one
+    reqPre.push_back("http://");
+    reqPre.push_back("https://");
+
+    // check fronts
+    bool found = false;
+    for (int i=0; i < reqPre.size() && !found; i++) {
+        if (strURL.find(reqPre[i]) == 0) found = true;
```

Comment:
minor nit: it would be nice to add a `break` after `found = true`.
Not really something that needs to be addressed now, since we have to check just for two cases (but something to keep in mind if later the prefixes list gets expanded).

Reply:
Ahh yes; good catch.

Reply:
whops nevermind... I see there's `i < reqPre.size() && !found;` in the cycle condition. :)

Reply:
lol!  I did find it surprising that I would have let something like that slip by.  I guess I didn't :)